narwhals/_pandas_like/expr.py

```diff
@@ -387,19 +387,66 @@ def alias(self, name: str) -> Self:
         )
 
     def over(self, keys: list[str]) -> Self:
-        def func(df: PandasLikeDataFrame) -> list[PandasLikeSeries]:
-            if self._output_names is None:
-                msg = (
-                    "Anonymous expressions are not supported in over.\n"
-                    "Instead of `nw.all()`, try using a named expression, such as "
-                    "`nw.col('a', 'b')`\n"
+        cumulative_functions_to_pandas_equivalent = {
+            "col->cum_sum": "cumsum",
+            "col->cum_min": "cummin",
+            "col->cum_max": "cummax",
+            "col->cum_prod": "cumprod",
+            # Pandas cumcount starts counting from 0 while Polars starts from 1
+            # Pandas cumcount counts nulls while Polars does not
+            # So, instead of using "cumcount" we use "cumsum" on notna() to get the same result
+            "col->cum_count": "cumsum",
+        }
+
+        if self._function_name in cumulative_functions_to_pandas_equivalent:
+
+            def func(df: PandasLikeDataFrame) -> list[PandasLikeSeries]:
+                if self._output_names is None:
+                    msg = (
+                        "Anonymous expressions are not supported in over.\n"
+                        "Instead of `nw.all()`, try using a named expression, such as "
+                        "`nw.col('a', 'b')`\n"
+                    )
+                    raise ValueError(msg)
+                if self._root_names is None:
+                    msg = (
+                        "Anonymous expressions are not supported in over with cumulative "
+                        "operations.\n"
+                    )
+                    raise ValueError(msg)
+
+                if self._function_name == "col->cum_count":
+                    df = df.with_columns(~df[self._root_names[0]].is_null())
```

Comment:
To have it work with multiple columns you could access narwhals namespace and use expressions:

```suggestion
                if self._function_name == "col->cum_count":
                    plx = self.__narwhals_namespace__()
                    df = df.select(~plx.col(self.root_names).is_null())
```

Reply:
i still needs to be `with_columns` right though? `keys` still needs to be part of the dataframe

Reply:
Yes absolutely! At first I wanted to select group keys and cum count columns, and then I forgot to switch back on the suggestion

Reply:
Thank a lot for your help @FBruzzesi ! 

I now have:

```python
df = df.with_columns(
    *[~df[name].is_null() for name in self._root_names]
)
```

Which I believe it should do the same as you suggested (not 100% sure, please correct me if I am wrong)

I tried with your suggestion first and I got the following error:

```python
AttributeError: 'DataFrame' object has no attribute 'name'
```

I did not really understand what's causing it tbh.